components/ILIAS/Chatroom/classes/class.ilChatroomAppEventListener.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+class ilChatroomAppEventListener implements ilAppEventListener
+{
+    /**
+     * @param array<string,mixed>  $a_parameter
+     */
+    public static function handleEvent(string $component, string $event, array $parameter): void
+    {
+        if ($event === 'deleteUser') {
+            global $DIC;
```

Comment:
IMO we should avoid database communication in listeners (I know there will be many examples where it is done this way).

Reply:
Should I add a `static` method in the `ilChatroom` class instead?! (non static doesn't make sense IMO as this is not specific to one room but all of them).

Reply:
IMO the best approach is to add a new service class.

Reply:
Done, please also note that I changed the SQL statement.